src/main/java/net/dv8tion/jda/core/requests/restaction/order/ChannelOrderAction.java

```diff
@@ -84,6 +84,16 @@ public ChannelOrderAction(Guild guild, ChannelType type)
         this.orderList.addAll(chans);
     }
 
+    // Internal usage only.
+    // This is used as a superclass constructor call for CategoryOrderAction.
+    ChannelOrderAction(Guild guild, ChannelType type, Collection<T> channels)
```

Comment:
This can be public

Reply:
Can the other constructor use this as well for centralized checks?

Reply:
I need some elaboration on this.
What checks need to be centralized exactly?
There are no checks made in the original constructor, unless you want to count the IllegalArgumentException default case as a check, which in that case I am not entirely sure how I am supposed to centralize the check made there since it is naturally part of the switch and therefore couldn't be moved.
Or are you referring to `CategoryOrderAction`s constructor moving it's checks to the superconstructor in `ChannelOrderAction`?

Reply:
There are checks (like null-checks) that need to happen on every creation. This constructor should be the default constructor that both CategoryOrderAction(Category, ChannelType) and ChannelOrderAction(Guild, ChannelType) would use.